internal/ebutil/libs.go

```diff
@@ -0,0 +1,78 @@
+package ebutil
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+)
+
+// Based on https://github.com/NVIDIA/libnvidia-container/blob/v1.15.0/src/common.h#L29
+
+const usrLibDir = "/usr/lib64"
+
+const debianVersionFile = "/etc/debian_version"
+
+// getLibDir returns the library directory. It returns a multiarch directory if
+// the distribution is Debian or a derivative.
+//
+// Based on https://github.com/NVIDIA/libnvidia-container/blob/v1.15.0/src/nvc_container.c#L152-L165
+func getLibDir(m mounter) (string, error) {
+	// Debian and its derivatives use a multiarch directory scheme.
+	if _, err := m.Stat(debianVersionFile); err != nil && !errors.Is(err, os.ErrNotExist) {
+		return "", fmt.Errorf("check if debian: %w", err)
+	} else if err == nil {
+		return usrLibMultiarchDir, nil
+	}
+
+	return usrLibDir, nil
+}
+
+// getLibsSymlinks returns the stats for all library symlinks if the library
+// directory exists.
+func getLibsSymlinks(m mounter, libDir string) (map[string][]string, error) {
```

Comment:
I'm curious if we need a map here. Can we flatten the structure?

Reply:
The mapping `library` (e.g. `libcuda.so.550.54.15`, what could be a mountpoint) -> `symlinks` (e.g. `[libcuda.so.1, libcuda.so]`). The function comment was outdated, I have updated it.